packages/picasso-query-builder/src/utils/get-query-depth.ts

```diff
@@ -0,0 +1,9 @@
+import type { RuleGroupTypeAny } from 'react-querybuilder'
+
+export const getQueryDepth = (arr: RuleGroupTypeAny): number => {
+  return Array.isArray(arr)
+    ? 1 + Math.max(...arr.map(getQueryDepth))
+    : typeof arr === 'object'
+    ? Math.max(...Object.values(arr).map(getQueryDepth))
+    : 0
+}
```

Comment:
```suggestion
export const getQueryDepth = (arr: RuleGroupTypeAny): number => {
  if (Array.isArray(arr)) {
    return 1 + Math.max(...arr.map(getQueryDepth));
  }

  if (typeof arr === 'object') {
    return Math.max(...Object.values(arr).map(getQueryDepth));
  }

  return 0;
}
```

Maybe it is only me, but even this function is pretty small, it is hard to read.

Reply:
Yes i do not like the double ternary operator either.